File: solutions/zig/collatz-conjecture/1/collatz_conjecture.zig
// Please implement the `ComputationError.IllegalArgument` error.

pub const ComputationError = error{
    IllegalArgument,
};

pub fn steps(number: usize) anyerror!usize {
    var number2 = number;
    if (number2 <= 0) {
        return ComputationError.IllegalArgument;
    }
    var count: usize = 0;
    while (number2 != 1) {
        if (number2 % 2 == 0) {
            number2 = number2 / 2;
        } else {
            number2 = (number2 * 3) + 1;
        }
        count += 1;
    }
    return count;
}

File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(n int) (int, error) {
    if n <= 0{
        return -1, errors.New("faield")
    }
	count := 0
    for n != 1{
        if n & 1 != 1{
            n = n/2
        }else{
            n = 3*n + 1
        }
        count += 1
    }
    return count,nil
}
